pkg/gatewayroute/resource_manager.go

```diff
@@ -110,7 +110,10 @@ func (m *defaultResourceManager) Cleanup(ctx context.Context, gr *appmesh.Gatewa
 		return err
 	}
 	sdkGR, err := m.findSDKGatewayRoute(ctx, ms, vg, gr)
-	if sdkGR == nil {
+	if err != nil {
+		if sdkGR == nil || gr.Status.GatewayRouteARN == nil {
```

Comment:
`sdkGR == nil` check is also applicable for case when `err == nil`, I'd suggest checking this outside the `err != nil` statement so it covers for nil `sdkGR` regardless of error

Same applies to virtual gateway `sdkVG `